kafka/kafka.go

```diff
@@ -167,6 +167,7 @@ type TopicPartition struct {
 	Topic     *string
 	Partition int32
 	Offset    Offset
+	Metadata  *string
```

Comment:
Metadata is defined as a binary blob in the protocol, so change this to byte type

Reply:
Perhaps I misunderstood, but in protocol (OffsetCommit Request (Version: 6)) metadata field defined as NULLABLE_STRING https://kafka.apache.org/protocol

Reply:
Sorry, I was mistaken, you're right!